cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -386,17 +391,165 @@ private KafkaRebalanceStatus buildRebalanceStatus(String sessionID, KafkaRebalan
+                    brokerStats.put(intLoadParameter.getStrimziKey(), intStats);
+                } else {
+                    log.warn("{} information was missing from the broker before/after load information",
+                            intLoadParameter.getStrimziKey());
+                }
+
+            }
+
+            for (CruiseControlLoadParameters doubleLoadParameter : CruiseControlLoadParameters.getDoubleParameters()) {
+
+                if (brokerBefore.containsKey(doubleLoadParameter.getStrimziKey()) &&
+                        brokerAfter.containsKey(doubleLoadParameter.getStrimziKey())) {
+
+                    double doubleBeforeStat = (double) brokerBefore.get(doubleLoadParameter.getStrimziKey());
+                    double doubleAfterStat = (double) brokerAfter.get(doubleLoadParameter.getStrimziKey());
+                    double doubleDiff = doubleAfterStat - doubleBeforeStat;
+
+                    double[] doubleStats = {doubleBeforeStat, doubleAfterStat, doubleDiff};
+                    brokerStats.put(doubleLoadParameter.getStrimziKey(), doubleStats);
+                } else {
+                    log.warn("{} information was missing from the broker before/after load information",
+                            doubleLoadParameter.getStrimziKey());
+                }
+
+            }
+
+            brokersBeforeAfterStats.put(brokerID, brokerStats);
+        }
+
+        return brokersBeforeAfterStats;
+
+    }
+
+    /**
+     * Converts the supplied JSONObject containing the response from the {@link CruiseControlApi#rebalance} or
+     * {@link CruiseControlApi#getUserTaskStatus} methods, into a map linking to a proposal summary map and a broker
+     * load map.
+     *
+     * @param  proposalJson The JSONObject representing the response from the Cruise Control rebalance endpoint.
+     * @return A map containing the proposal summary and broker load maps.
+     */
+    protected static Map<String, Object> processOptimizationProposal(JsonObject proposalJson) {
+
+        JsonArray brokerLoadBeforeOptimization;
+        JsonArray brokerLoadAfterOptimization;
+        if (proposalJson.containsKey(CruiseControlRebalanceKeys.LOAD_BEFORE_OPTIMIZATION.getKey()) &&
+                proposalJson.containsKey(CruiseControlRebalanceKeys.LOAD_AFTER_OPTIMIZATION.getKey())) {
+            brokerLoadBeforeOptimization = proposalJson
+                    .getJsonObject(CruiseControlRebalanceKeys.LOAD_BEFORE_OPTIMIZATION.getKey())
+                    .getJsonArray(CruiseControlRebalanceKeys.BROKERS.getKey());
+            brokerLoadAfterOptimization = proposalJson
+                    .getJsonObject(CruiseControlRebalanceKeys.LOAD_AFTER_OPTIMIZATION.getKey())
+                    .getJsonArray(CruiseControlRebalanceKeys.BROKERS.getKey());
+        } else {
+            throw new RuntimeException("The rebalance optimization proposal returned by Cruise Control did not contain broker load information");
```

Comment:
Is there any reason these keys may be missing, and if so might we want to fall back to the old behaviour which didn't assume they exist?

I think the answer is no, but just a question I thought worth bringing up.

Reply:
It is a good point, these keys are probably unlikely to be missing as anything that would cause that is likely to cause a 500 error on the CC server. 

On the client side we use the presence of the summary key to tell weather the proposal was successful so we know it will be there when we reach this point, but in an over abundance of caution I checked the other keys. 

Is this overkill? I could check for key presence in the client and only return if they are all present?

Reply:
I don't mind too much either way, I think you're right that these are very unlikely to ever be empty. so happy to approve.